fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/models/ResourceFactoryImpl.java

```diff
@@ -109,47 +83,66 @@ public FedoraResource getResource(final String identifier)
     @Override
     public FedoraResource getResource(final Transaction transaction, final String identifier)
             throws PathNotFoundException {
-        try {
-            final PersistentStorageSession psSession = getSession(transaction);
-            return psSession.read(identifier);
-        } catch (final PersistentItemNotFoundException e) {
-            throw new PathNotFoundException(e);
-        } catch (final PersistentStorageException e) {
-            // This is a big error, wrap as RepositoryRuntime and send it through.
-            throw new RepositoryRuntimeException(e);
-        }
+        return createResource(transaction, identifier);
```

Comment:
Why does getResource() call createResource()?  Is it creating a new one or simply creating a new object from information found at the persistence layer?  Thanks!

Reply:
@bseeger :  my understanding of the shift is that FedoraResources are now being built and populated above the persistence layer.  So I think the latter is @bbpennel 's intention.

Reply:
The terminology may be a little confusing. This method is creating a FedoraResource object, rather than persisting its contents. I could rename it to `instantiateResource` or something similar if that would be clearer.

Reply:
or `fetchResource`?

Reply:
`instantiateResource` would probably work better then fetch.

Reply:
https://github.com/fcrepo4/fcrepo4/pull/1557/commits/f14c3aad76ba972d2e7ce1aad5fc347749209726 has got you covered